ncloud/resource_ncloud_login_key.go

```diff
@@ -80,96 +78,174 @@ func resourceNcloudLoginKeyUpdate(d *schema.ResourceData, meta interface{}) erro
 
 	return resourceNcloudLoginKeyRead(d, meta)
 }
 
 func resourceNcloudLoginKeyDelete(d *schema.ResourceData, meta interface{}) error {
-	client := meta.(*ProviderConfig).Client
+	config := meta.(*ProviderConfig)
 
 	keyName := d.Get("key_name").(string)
 
-	if err := deleteLoginKey(client, keyName); err != nil {
-		return err
+	if config.SupportVPC == true || config.Site == "fin" {
+		if err := deleteVpcLoginKey(config.Client, &keyName); err != nil {
+			return err
+		}
+	} else {
+		if err := deleteClassicLoginKey(config.Client, &keyName); err != nil {
+			return err
+		}
 	}
+
 	d.SetId("")
+
 	return nil
 }
 
-func getLoginKeyList(client *NcloudAPIClient, keyName *string) (*server.GetLoginKeyListResponse, error) {
+func getClassicFingerPrintList(client *NcloudAPIClient, keyName *string) ([]*string, error) {
 	reqParams := &server.GetLoginKeyListRequest{}
 	if keyName != nil {
 		reqParams.KeyName = keyName
 	}
 
-	logCommonRequest("GetLoginKeyList", reqParams)
-
+	logCommonRequest("getClassicFingerPrintList", reqParams)
 	resp, err := client.server.V2Api.GetLoginKeyList(reqParams)
 	if err != nil {
-		logErrorResponse("GetLoginKeyList", err, reqParams)
+		logErrorResponse("getClassicFingerPrintList", err, reqParams)
 		return nil, err
 	}
+	logResponse("getClassicFingerPrintList", resp)
 
-	var totalRowsLog string
-	if resp != nil {
-		totalRowsLog = fmt.Sprintf("totalRows: %d", ncloud.Int32Value(resp.TotalRows))
+	keyList := make([]*string, 0, *resp.TotalRows)
+	for _, v := range resp.LoginKeyList {
+		keyList = append(keyList, v.Fingerprint)
 	}
-	logCommonResponse("GetLoginKeyList", GetCommonResponse(resp), totalRowsLog)
-	return resp, nil
+
+	return keyList, nil
 }
 
-func getLoginKey(client *NcloudAPIClient, keyName string) (*server.LoginKey, error) {
-	resp, err := getLoginKeyList(client, ncloud.String(keyName))
-	if len(resp.LoginKeyList) > 0 {
-		return resp.LoginKeyList[0], err
+func getVpcFingerPrintList(client *NcloudAPIClient, keyName *string) ([]*string, error) {
+	reqParams := &vserver.GetLoginKeyListRequest{}
+	if keyName != nil {
+		reqParams.KeyName = keyName
+	}
+
+	logCommonRequest("getVpcFingerPrintList", reqParams)
+	resp, err := client.vserver.V2Api.GetLoginKeyList(reqParams)
+	if err != nil {
+		logErrorResponse("getVpcFingerPrintList", err, reqParams)
+		return nil, err
+	}
+	logResponse("getVpcFingerPrintList", resp)
+
+	keyList := make([]*string, 0, *resp.TotalRows)
+	for _, v := range resp.LoginKeyList {
+		keyList = append(keyList, v.Fingerprint)
 	}
 
-	return nil, err
+	return keyList, nil
 }
 
-func deleteLoginKey(client *NcloudAPIClient, keyName string) error {
-	reqParams := &server.DeleteLoginKeyRequest{KeyName: ncloud.String(keyName)}
-	logCommonRequest("DeleteLoginKey", reqParams)
+func getFingerPrint(config *ProviderConfig, keyName *string) (*string, error) {
+	var fingerPrintList []*string
+	var err error
 
+	if config.SupportVPC == true || config.Site == "fin" {
```

Comment:
```suggestion
	if config.SupportVPC == true {
```
provider.go 로직을 보면 `config.Site == "fin"` 이면 `providerConfig.SupportVPC = true` 로 설정이 되어
`config.SupportVPC`만 체크해도 좋을것 같습니다.